twine/wheel.py

```diff
@@ -62,10 +49,7 @@ def py_version(self):
     @staticmethod
     def find_candidate_metadata_files(names):
         """Filter files that may be METADATA files."""
-        tuples = [
-            x.split('/') for x in map(try_decode, names)
-            if 'METADATA' in x
-        ]
+        tuples = [x.split('/') for x in names if 'METADATA' in x]
```

Comment:
This function previously would have supported bytes as input, but does no longer. Can we say with confidence that there's no Python 3 usage that might be passing bytes for names?

Reply:
Yes, I believe we can now assume this is always text. This is directly related to the changes to [test_wheel.py](https://github.com/pypa/twine/pull/437#discussion_r246014184).

The only call to `find_candidate_metadata_files()` is in `twine/wheel.py`.

https://github.com/pypa/twine/blob/5a72af19fdb27d402c7bcea41199e5eae9082fbb/twine/wheel.py#L90

The arguments to this call always come from `ZipFile.namelist()` (a few lines up in the link).

https://github.com/pypa/twine/blob/5a72af19fdb27d402c7bcea41199e5eae9082fbb/twine/wheel.py#L80

On Python 2, `ZipFile.namelist()` can return either `str` (bytes type) or `unicode`. Here is the relevant CPython code:

https://github.com/python/cpython/blob/v2.7.15/Lib/zipfile.py#L377-L381

On Python 3, `ZipFile.namelist()` always returns a `str` (text type). Here is the relevant CPython code:

https://github.com/python/cpython/blob/v3.7.2/Lib/zipfile.py#L1322-L1327

So in summary, we can now assume the value never needs to be decoded and the test can be altered to `str` (not `bytes`) to be more representative of `ZipFile`.